src/Symfony/Component/DependencyInjection/Compiler/ResolveInvalidReferencesPass.php

```diff
@@ -101,6 +103,12 @@ private function processValue($value, int $rootLevel = 0, int $level = 0)
             if ($this->container->has($id = (string) $value)) {
                 return $value;
             }
+
+            // resolve decorated service behavior depending on decorator service
+            if (ContainerInterface::NULL_ON_INVALID_REFERENCE === $this->currentDefinition->onDecoratedInvalid) {
```

Comment:
this looks wrong to me, as the decoration setting should apply only to the reference to the inner service, not to all references of the decorator service

Reply:
You're perfectly right !
I fixed it and added related test, thanks.